### PR TITLE
builders: fix author position sorting

### PR DESCRIPTION
**Pull request by harunurhan**
* Fixes the sorting when there are positions with and without start_date

* Fixes the sorting when there are positions with and without rank

### CODE REVIEW

File: inspire_schemas/builders/authors.py [hunk 1]
```diff
@@ -34,7 +34,9 @@
 
 
 RANKS = load_schema('elements/rank')['enum']
+RANKS.append(None)
 INSTITUTION_RANK_TO_PRIORITY = {rank: -idx for (idx, rank) in enumerate(RANKS)}
+EARLIEST_DATE = PartialDate.parse('1-1-1000')
```
Comment (michamos) [inline inspire_schemas/builders/authors.py]: `.loads('1000-01-01')` so you don't need to go through the parser.

**Comment by harunurhan** [comment]
@michamos adressed!

File: inspire_schemas/builders/authors.py [hunk 2]
```diff
@@ -34,7 +34,9 @@
 
 
 RANKS = load_schema('elements/rank')['enum']
+RANKS.append(None)
```
Comment (michamos) [inline inspire_schemas/builders/authors.py]: why is this needed?

Comment (harunurhan) [reply inspire_schemas/builders/authors.py]: if there is one with rank and one without, it fails. But it should be `any rank > no rank`.

Comment (harunurhan) [reply inspire_schemas/builders/authors.py]: maybe better to put a comment for this.
